morpheus/_lib/include/morpheus/messages/multi.hpp

```diff
@@ -79,10 +80,30 @@ class MultiMessage
     void set_meta(const std::vector<std::string> &column_names, const std::vector<TensorObject> &tensors);
 
     /**
-     * TODO(Documentation)
+     * @brief Creates a copy of the current message calculating new `mess_offset` and `mess_count` values based on the
+     * given `start` & `stop` values. This method is reletively light-weight as it does not copy the underlying `meta`
+     * and the actual slicing of the dataframe is applied later when `get_meta` is called.
+     *
+     * @param start
+     * @param stop
+     * @return std::shared_ptr<MultiMessage>
      */
     std::shared_ptr<MultiMessage> get_slice(size_t start, size_t stop) const;
 
+    /**
+     * @brief Creates a deep copy of the current message along with a copy of the underlying `meta` selecting the rows
+     * of `meta` defined by pairs of start, stop rows expressed in the `ranges` argument.
+     *
+     * This allows for copying several non-contiguous rows from the underlying dataframe into a new dataframe, however
+     * this comes at a much higher cost compared to the `get_slice` method.
+     *
+     * @param ranges
+     * @param num_selected_rows
+     * @return std::shared_ptr<MultiMessage>
+     */
+    std::shared_ptr<MultiMessage> copy_ranges(const std::vector<std::pair<size_t, size_t>> &ranges,
```

Comment:
> We will eventually need a way to copy ranges from multiple multimessages in to a single contiguous range, and create one or more multimessages as views over that range.

I'm not sure we will need that as `on_next` is invoked once per message and nodes are intended to operate on one message at a time.

> is num_selected_rows the sum of std::get<1>(...)-std::get<0>(...) of each pair? If so, why don't we just compute that inside the function?

Yes, the reason is that the caller already has that number, and this is to avoid the cost of re-calculating it.